src/server/ProcessHandle.cpp

```diff
@@ -34,6 +37,44 @@ ConsoleProcessHandle::ConsoleProcessHandle(const DWORD dwProcessId,
     {
         Telemetry::Instance().LogProcessConnected(_hProcess.get());
     }
+
+    // GH#13211 - If we're running as the delegation console (someone handed off
+    // to us), then we need to make sure the original conhost has access to this
+    // process handle as well. Otherwise, the future calls to
+    // ConsoleControl(SetForeground,...) won't work, because the literal handle
+    // value doesn't exist in the original conhost process space.
+    // * g.handoffInboxConsoleHandle is only set when we've been delegated to
+    // * We can't just pass something like the PID, because the OS conhost
+    //   already expects a literal handle value via the HostSignalInputThread.
+    //   If we changed that in the OpenConsole version, then there'll surely be
+    //   the chance for a mispatch between the OS conhost and the delegated
+    //   console.
+    if (const auto& conhost{ ServiceLocator::LocateGlobals().handoffInboxConsoleHandle })
+    {
+        LOG_IF_WIN32_BOOL_FALSE(DuplicateHandle(GetCurrentProcess(),
+                                                _hProcess.get(),
+                                                conhost.get(),
+                                                _hProcessInConhost.put(),
+                                                0 /*dwDesiredAccess, ignored*/,
+                                                false,
+                                                DUPLICATE_SAME_ACCESS));
+    }
+}
+
+ConsoleProcessHandle::~ConsoleProcessHandle()
+{
+    // Close out the handle in the origin conhost.
+    const auto& conhost{ ServiceLocator::LocateGlobals().handoffInboxConsoleHandle };
+    if (_hProcessInConhost && conhost)
+    {
+        LOG_IF_WIN32_BOOL_FALSE(DuplicateHandle(conhost.get(),
+                                                _hProcessInConhost.get(),
+                                                nullptr /* hTargetProcessHandle */,
+                                                nullptr /* lpTargetHandle, ignored */,
+                                                0 /* dwDesiredAccess, ignored */,
+                                                false,
+                                                DUPLICATE_CLOSE_SOURCE));

```

Comment:
HAAHAHAHAHA OH GOD


I read your description and was all like, "Are we about to cause a handle leak in conhost"

Here I see that my concerns were unfounded

Reply:
nah see I'm a good programmer (on Wednesdays) 😄

Reply:
lmao I was a bad programmer that Wednesday 🙃